### ISSUE TITLE
Kuma should stop setting heading id attributes to match name attributes

### ISSUE THREAD

**Issue by wbamberg** [mdn/kuma#7389]
In Kuma, headings automatically get `id` attributes that are taken from the heading's text content:

```
<h2 id="Examples">Examples</h2>
```

Unless the heading also has a `name` attribute, in which case the value of the `name` is used as the `id`:

```
<h2 id="Example" name="Example">Examples</h2>
```

This is a problem for writers, because the linter relies on heading IDs to, well, identify headings. And one of the things we are doing as part of the linting work is to fix up our headings to make them consistent, so we're (manually) making hundreds of changes like:

```
"Example" -> "Examples"
"Browser Compatibility" -> "Browser compatibility" 
"Specification" -> "Specifications"
```

Because of Kuma's treatment of the `name` attribute, making these changes sometimes doesn't update the `id`, so the linter still reports an error, and we have to go into the source view and remove the `name` attribute manually.

I don't know of any good reason for Kuma's behaviour here, so I'd like to suggest that it is changed so `id` is set only from the text content, and the `name` attribute is ignored.

**Comment by escattone** [comment]
@wbamberg This topic fills me with some fear. One complication that comes to mind is that the `name` attribute ensures consistency for the `EmbedLiveSample` macro calls in terms of what `id` to use for the section holding the live sample. For example, https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Traversing_an_HTML_table_with_JavaScript_and_DOM_Interfaces contains the following section header for a live sample:
```html
<h2 id="Overview_of_Sample1.html" name="Overview_of_Sample1.html">Example: Creating an HTML table dynamically (<code>Sample1.html</code>)</h2>
...
<p>{{ EmbedLiveSample('Overview_of_Sample1.html') }}</p>
```

If there were no `name` attribute, the code that converts the heading text to an `id` would create `Example_Creating_an_HTML_table_dynamically_Sample1.html` which is difficult to predict (unless the content editor knows the algorithm, which strips out a specific set of characters) when selecting the argument for the `EmbedLiveSample` macro call.

Also, for translations, the translators often translate the heading text, for example:
```html
<h2 id="Overview_of_Sample1.html" name="Overview_of_Sample1.html">Ejemplo: Crear una tabla HTML dinámicamente (<code>Ejemplo1.html</code>)</h2>
```

but without a `name` attribute they'd also have to remember to modify the argument to the `EmbedLiveSample` macro call as well.

**Comment by wbamberg** [comment]
This would be a more powerful argument if `name` were consistently used for this purpose as part of a designed plan, rather than being an occasional possible side-effect of a feature that apparently no longer has a primary purpose.

I count 5286 H2 headings in the CSS docs, of which 311 have the `name` attribute set. So there's a about a 5% chance that it would save someone from breaking a live sample by editing a title.

But, I'm sympathetic to your desire to not want to touch the code any more than you have to at this point, and your fear of the side-effects that any change might bring.

**Comment by escattone** [comment]
@wbamberg Yes, this area feels like a mess. There's no consistency in approach. There are two issues here as I see it:
1. We don't just use the text "as is" to create the `id` attribute, but instead remove "illegal" characters and replace whitespace with underscores, so it's not always easy to predict the `id` in some cases when the heading text contains "illegal" characters. An alternative approach would be to escape the text "as is" when creating the `id` attribute. Then the content creator/editor knows that its corresponding `EmbedLiveSample` macro call, if there is one, should always use the exact same text.
1. There's no policy on what translations should do. If we remove this current approach of allowing the special `name` attribute, all translations that localize the heading text would also have to localize the corresponding `EmbedLiveSample` macro call, if there is one.

We discussed this within our bug triage meeting this morning, and thought that it's probably best to avoid making any changes for now, and then revisit this once we've released Yari, when we'll have control over every part of the problem. Then we can more easily arrive at a consistent approach. If you disagree @wbamberg and feel that this is more urgent, I'm happy to discuss it further.
